.github/workflows/build.yml

```diff
@@ -0,0 +1,29 @@
+name: 'build'
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        idf_ver: ["v4.1", "v4.2", "v4.3"]
+        idf_target: ["esp32"]
+        include:
+          - idf_ver: "v4.2"
+            idf_target: esp32s2
+            # TODO: enable C3 build
+          #- idf_ver: "v4.3"
+          #  idf_target: esp32c3
+    runs-on: ubuntu-20.04
+    container: espressif/idf:release-${{ matrix.idf_ver }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Build for ESP32
+        env:
+          IDF_TARGET: ${{ matrix.idf_target }}

```

Comment:
@tore-espressif I have a silly question, if we add more targets into the `idf_target` "vector" (can we call it vector?) the job will run on all the combinations between `idf_ver` and `idf_target`?

Reply:
Exactly, as it is now, it would run 3x for ESP32 (IDF v 4.1, 4.2 and 4.3) and once for ESP32-S2 (v4.2). So adding new IDF versions/target should be intuitive.

I removed version 4.0 and 3.3 because it would fail on `rsource` in Kconfig, which is supported from v4.1 onwards. https://github.com/lvgl/lvgl_esp32_drivers/blob/9fed1cc47b5a45fec6bae08b55d2147d3b50260c/Kconfig#L1.

Reply:
Thanks for the reply, let us know when you want to merge this.